player/loadfile.c

```diff
@@ -832,8 +832,10 @@ int mp_add_external_file(struct MPContext *mpctx, char *filename,
         t->external_filename = talloc_strdup(t, filename);
         t->no_default = sh->type != filter;
         t->no_auto_select = t->no_default;
-        // filter==STREAM_VIDEO always means cover art.
-        t->attached_picture = t->type == STREAM_VIDEO && filter == STREAM_VIDEO;
+        // filter==STREAM_VIDEO does *NOT* always means cover art.
+        // check filename to assume if image or video.
+        t->attached_picture = t->type == STREAM_VIDEO && filter == STREAM_VIDEO &&
+                              mp_might_be_image_file(filename);
```

Comment:
Removing the these two lines would also fix the bug. `attached_picture` is already set here for cover files:
https://github.com/mpv-player/mpv/blob/c766e47b7095568509173e49fb049ff1b501b71d/player/loadfile.c#L907-L908

Reply:
I know it's doing a similar check twice with slightly different methods, but my way will also detect artwork if an image is passed in the video-add command, so it'll treat a picture like a picture, not a video.

Reply:
The reason I'm suggesting a different approach is that probing the filename is prone to misdetection:
* listing all possible image formats is not feasible
* if you add an MJPEG stream ending with that extension (like `http://192.168.1.20/video.jpg`) this also breaks
* it cannot correctly account for special protocols such as `memory://`

Since video-add was never specified to support attaching picture streams I'd prefer to reserve that behavior for files added by the coverart code.

Reply:
OK, that makes sense.
But if an image is added with video-add will it matter if the resulting video track has attached_picture=false? I'm assuming it plays fine but is less efficiently handled. If that's the case I'll do it your way and re-commit.

Reply:
That's what I expect to happen, yes.

Reply:
This pull request may not be the place to implement it, but an alternative might be to create an extra flag for the video-add command to treat the given file as a picture.